File: pinyougou-manager-web/src/main/webapp/js/pinyougou/controller/brandController.js
app.controller("brandController",function($scope,$controller,brandService){

    /*继承父模块*/
    $controller("baseController",{$scope:$scope});

    $scope.findAll=function(){
        brandService.findAll().success(function (responseData) {
            $scope.brandList=responseData;
        });
    }

    $scope.searchEntity={};
    /*分页查询*/
    $scope.findPage=function(page,rows){
        brandService.findPage(page,rows,$scope.searchEntity).success(function(responseData){
            $scope.brandList=responseData.rows;
            //更新总记录数
            $scope.paginationConf.totalItems=responseData.total;
        });
    }

    /*保存*/
    $scope.save=function(){
        brandService.save($scope.brand).success(function(responseData){
            if (responseData.success){
                alert(responseData.message);
                $scope.reloadList();
            }else{
                alert(responseData.message);
            }
        });
    };

    /*查询单个品牌信息*/
    $scope.findOne=function(id){
        brandService.findOne(id).success(function(responseData){
            $scope.brand=responseData;
        });
    };

    /*批量删除*/
    $scope.deleteBatch=function(){
        brandService.deleteBatch($scope.selectIds).success(function(responseData){
            if (responseData.success){
                alert(responseData.message);
                $scope.reloadList();
                $scope.selectIds=[];
            }else{
                alert(responseData.message);
            }
        });
    };
});

File: pinyougou-manager-web/src/main/java/com/rogercw/pinyougou/web/controller/BrandController.java
package com.rogercw.pinyougou.web.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.rogercw.page.PageResult;
import com.rogercw.pinyougou.pojo.Brand;
import com.rogercw.pinyougou.service.BrandService;
import com.rogercw.result.ResponseMsg;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/**
 * @Author: rogercw
 * @Date: 2018/8/29 23:15
 * @Version 1.0
 */
@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;

    @RequestMapping("/findAll")
    public List<Brand> findAll(){
        return brandService.findAll();
    }

    @RequestMapping("/findPage")
    public PageResult findPage(int page,int rows,@RequestBody Brand brand){
        return brandService.findPage(page, rows,brand);
    }


    @RequestMapping("/save")
    public ResponseMsg save(@RequestBody Brand brand){
        try {
            if (brand.getId()==null){
                //新增
                brandService.add(brand);
            }else{
                brandService.update(brand);
            }
            return new ResponseMsg(true,"保存成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new ResponseMsg(false,"保存失败");
        }
    }

    @RequestMapping("/findOne")
    public Brand findOne(Long id){
        return brandService.findOne(id);
    }

    @RequestMapping("/delete")
    public ResponseMsg delete(Long[] ids){
        try {
            brandService.delete(ids);
            return new ResponseMsg(true,"删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new ResponseMsg(false,"删除失败");
        }
    }

}
